source/puppet/4.10/lang_data_number.md

```diff
@@ -57,49 +56,63 @@ $product = 8 * 3e5  # evaluates to 240000.0
+$value = 0777.3  # Error, invalid octal
+
+# Hexadecimal
+$value = 0x777  # evaluates to decimal 1911
+$value = 0xdef  # evaluates to decimal 3567
+$value = 0Xdef  # same as above
+$value = 0xDEF  # same as above
+$value = 0xLSD  # Error, invalid hex
 ```
 
 ## Converting numbers to strings
 
 Numbers are automatically converted to strings when interpolated into a string. The automatic conversion uses decimal (base 10) notation.
 
-If you need to convert numbers to non-decimal string representations, you can use [the `sprintf` function.](./function.html#sprintf)
+To convert numbers to non-decimal string representations, you can use the [`sprintf` function](./function.html#sprintf), or you can create a new string object with the `new()` function and [specify its format](./function.html#conversion-to-string).
 
 ## Converting strings to numbers
 
-The [arithmetic operators][arithmetic] will automatically convert strings to numbers.
+[Arithmetic operators][arithmetic] automatically convert strings to numbers.
 
 In all other contexts (resource attributes, function arguments, etc.), Puppet _won't_ automatically convert strings to numbers, but you can:
 
-* Add 0 to manually convert a string to a number. (For example, `$mystring = "85"; $mynum = 0 + $mystring`.)
-* Use [the `scanf` function](./function.html#scanf) to manually extract numbers from strings. This function can also account for surrounding non-numerical text.
+-   Create a new `Integer` or `Float` object using the string as a parameter, such as `$mynum = Integer($mystring)`. This invokes the type's `new()` function, which can convert objects of other types --- including Strings --- into [Integers](./function.html#conversion-to-integer) and [Floats](./function.html#conversion-to-float).
+
+    The default radix of this conversion is base 10, but you can specify others.
+-   Use [the `scanf` function](./function.html#scanf) to manually extract numbers from strings. This function can also account for surrounding non-numerical text.
+
+> **Deprecation note:** Puppet 4.10.2 deprecated numeric coercion, which silently and automatically converted strings into numbers by adding a number to a string (for example, `$mynum = '10' + 0` would result in `$mynum` being assigned the number 10.) This now logs a Warning, or generates an Error if `--strict` is enabled.
+>
+> To ensure that your code works with future versions of Puppet, create new instances of numeric objects to convert String values to numeric types.
+
+### Examples
+
+Converting a string to an integer and float:
 
+    $mystring = '10';                  # $mystring == '10'
+    $myinteger = Integer($mystring);   # $myinteger == 10
+    $myfloat = Integer($mystring);     # $myfloat == 10.0
+    $myhexint = Integer($mystring, 16) # $myhexint == 16; '10' in base 16
 
 ## The `Integer` data type
 
 The [data type][] of integers is `Integer`.
 
 By default, `Integer` matches whole numbers of any size (within the limits of available memory).
```

Comment:
That is a strange description of what an Integer matches - has nothing to do with the limits of available memory. An integer is specified to be a 64 bit value and that sets the limits of which whole numbers it can match.